src/libraries/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.cs

```diff
@@ -165,7 +165,17 @@ public override object Invoke(BindingFlags invokeAttr, Binder? binder, object?[]
                 arguments = CheckArguments(ref stackArgs, parameters, binder, invokeAttr, culture, ArgumentTypes);
             }
 
-            object retValue = InvokeCtorWorker(invokeAttr, arguments);
+            object retValue;
+
+            // Array construction is a special case.
+            if (DeclaringType is not null && DeclaringType.IsArray)
```

Comment:
Is moving the array check here affect performance in measurable way?

Reply:
I fixed the widening logic in the original path and reran the scenarios. I see very little impact on the change. Thoughts?

Old:
       230 ms: AllocateArray[100]: 1000000
       139 ms: DefaultConstructor: 1000000
       200 ms: IntConstructor: 1000000

New:
       228 ms: AllocateArray[100]: 1000000
       141 ms: DefaultConstructor: 1000000
       201 ms: IntConstructor: 1000000